### ISSUE TITLE
rvm 1.29.0 (master) - Somehow it happened there is no executable 'openssl'

### ISSUE THREAD

**Issue by peracchi** [rvm/rvm#3928]
Can't install ruby-2.4.0 on Mac OS X El Capitan 10.11.6 (15G1217)

```
 $ rvm version
rvm 1.29.0 (master) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io/]
 $ ruby -v
ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
 $ which ruby
/usr/bin/ruby
 $ gem -v
2.0.14.1
 $ which gem
/usr/bin/gem
 $ rvm install 2.4.0@learning --create --default
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.11/x86_64/ruby-2.4.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Installing requirements for osx.
Updating system......
Installing required packages: autoconf, automake, libtool, pkg-config, coreutils, libyaml, readline, openssl@1.1..........
Somehow it happened there is no executable 'openssl',
run 'brew doctor' and make sure latest '' is installed properly.
RVM autolibs is now configured with mode '4' =>
  'Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X).',
please run `rvm autolibs enable` to let RVM do its job or run and read `rvm autolibs [help]`
or visit https://rvm.io/rvm/autolibs for more information.
Requirements installation failed with status: 12.
 $ brew doctor
Your system is ready to brew.
 $ brew list
autoconf	coreutils	gnupg2		libassuan	libgpg-error	libtool		libusb-compat	openssl@1.1	pkg-config	readline
automake	dirmngr		gpg-agent	libgcrypt	libksba		libusb		libyaml		pinentry	pth		ssh-copy-id
 $ which openssl
/usr/bin/openssl
 $ rvm info

system:

  system:
    uname:        "Darwin samsara.local 15.6.0 Darwin Kernel Version 15.6.0: Mon Jan  9 23:07:29 PST 2017; root:xnu-3248.60.11.2.1~1/RELEASE_X86_64 x86_64"
    name:         "OSX"
    version:      "10.11"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)"
    zsh:          "/bin/zsh => zsh 5.0.8 (x86_64-apple-darwin15.0)"
    remote path:  "osx/10.11/x86_64"

  rvm:
    version:      "rvm 1.29.0 (master) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io/]"
    updated:      "14 minutes 4 seconds ago
    path:         "/Users/leandro/.rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  homes:
    gem:          "not set"
    ruby:         "not set"

  binaries:
    ruby:         "/usr/bin/ruby"
    irb:          "/usr/bin/irb"
    gem:          "/usr/bin/gem"
    rake:         "/usr/bin/rake"

  environment:
    PATH:         "/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/leandro/.rvm/bin"
    GEM_HOME:     ""
    GEM_PATH:     ""
    MY_RUBY_HOME: ""
    IRBRC:        ""
    RUBYOPT:      ""
    gemset:       ""
```

**Comment by mpapis** [comment]
try with:
```bash
rvm get branch /bugfix/3928_dynamic_openssl_check_for_ruby24
```

**Comment by peracchi** [comment]
@mpapis 

  Now everything worked flawlessly!

  Full commands output below:

```
leandro@samsara:~ $ rvm version
rvm 1.29.0 (master) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io/]
leandro@samsara:~ $ rvm get branch /bugfix/3928_dynamic_openssl_check_for_ruby24
Downloading https://get.rvm.io
Downloading https://raw.githubusercontent.com/wayneeseguin/rvm/master/binscripts/rvm-installer.asc
Verifying /Users/leandro/.rvm/archives/rvm-installer.asc
gpg: Signature made Wed Feb  8 09:38:29 2017 BRST using RSA key ID BF04FF17
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Subkey fingerprint: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/Users/leandro/.rvm/archives/rvm-installer'
Downloading https://github.com/rvm/rvm/archive/bugfix/3928_dynamic_openssl_check_for_ruby24.tar.gz

Upgrading the RVM installation in /Users/leandro/.rvm/
    Adding rvm PATH line to /Users/leandro/.profile /Users/leandro/.mkshrc /Users/leandro/.bashrc /Users/leandro/.zshrc.
    RVM sourcing line found in /Users/leandro/.bash_profile.
    RVM sourcing line not found for Zsh, rerun this command with '--auto-dotfiles' flag to fix it.
Upgrade of RVM in /Users/leandro/.rvm/ is complete.

# Leandro Peracchi,
#
#   Thank you for using RVM!
#   We sincerely hope that RVM helps to make your life easier and more enjoyable!!!
#
# ~Wayne, Michal & team.

In case of problems: https://rvm.io/help and https://twitter.com/rvm_io

Upgrade Notes:


  * WARNING: You have '~/.profile' file, you might want to load it,
    to do that add the following line to '/Users/leandro/.bash_profile':

      source ~/.profile

  * No new notes to display.

RVM reloaded!
leandro@samsara:~ $ rvm install 2.4.0@learning --create --default
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.11/x86_64/ruby-2.4.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates in '/usr/local/etc/openssl@1.1/cert.pem' are already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/leandro/.rvm/rubies/ruby-2.4.0, this may take a while depending on your cpu(s)...
ruby-2.4.0 - #downloading ruby-2.4.0, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 11.9M  100 11.9M    0     0  1485k      0  0:00:08  0:00:08 --:--:-- 1672k
ruby-2.4.0 - #extracting ruby-2.4.0 to /Users/leandro/.rvm/src/ruby-2.4.0....
ruby-2.4.0 - #configuring...................................................................
ruby-2.4.0 - #post-configuration.
ruby-2.4.0 - #compiling.............................................................
ruby-2.4.0 - #installing.......
ruby-2.4.0 - #making binaries executable..
ruby-2.4.0 - #downloading rubygems-2.6.10
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  747k  100  747k    0     0   995k      0 --:--:-- --:--:-- --:--:--  995k
ruby-2.4.0 - #extracting rubygems-2.6.10....
ruby-2.4.0 - #removing old rubygems.........
ruby-2.4.0 - #installing rubygems-2.6.10.....................
ruby-2.4.0 - #gemset created /Users/leandro/.rvm/gems/ruby-2.4.0@global
ruby-2.4.0 - #importing gemset /Users/leandro/.rvm/gemsets/global.gems...............................................
ruby-2.4.0 - #generating global wrappers........
ruby-2.4.0 - #gemset created /Users/leandro/.rvm/gems/ruby-2.4.0
ruby-2.4.0 - #importing gemsetfile /Users/leandro/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-2.4.0 - #generating default wrappers........
ruby-2.4.0 - #adjusting #shebangs for (gem irb erb ri rdoc testrb rake).
Install of ruby-2.4.0 - #complete
Ruby was built without documentation, to build it run: rvm docs generate-ri
ruby-2.4.0 - #gemset created /Users/leandro/.rvm/gems/ruby-2.4.0@learning
ruby-2.4.0 - #generating learning wrappers..........
leandro@samsara:~ $ rvm list

rvm rubies

=* ruby-2.4.0 [ x86_64 ]

# => - current
# =* - current && default
#  * - default

leandro@samsara:~ $ rvm gemset list

gemsets for ruby-2.4.0 (found in /Users/leandro/.rvm/gems/ruby-2.4.0)
   (default)
   global
=> learning

leandro@samsara:~ $ ruby -v
ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin15]
leandro@samsara:~ $ which ruby
/Users/leandro/.rvm/rubies/ruby-2.4.0/bin/ruby
leandro@samsara:~ $ gem -v
2.6.10
leandro@samsara:~ $ which gem
/Users/leandro/.rvm/rubies/ruby-2.4.0/bin/gem
leandro@samsara:~ $ gem list

*** LOCAL GEMS ***

bigdecimal (default: 1.3.0)
bundler-unload (1.0.2)
did_you_mean (1.1.0)
executable-hooks (1.3.2)
gem-wrappers (1.2.7)
io-console (default: 0.4.6)
json (default: 2.0.2)
minitest (5.10.1)
net-telnet (0.1.1)
openssl (default: 2.0.2)
power_assert (0.4.1)
psych (default: 2.2.2)
rake (12.0.0)
rdoc (default: 5.0.0)
rubygems-bundler (1.4.4)
rvm (1.11.3.9)
test-unit (3.2.3)
xmlrpc (0.2.1)
leandro@samsara:~ $ gem update --system
Latest version currently installed. Aborting.
leandro@samsara:~ $ gem outdated
bigdecimal (1.3.0 < 1.3.1)
json (2.0.2 < 2.0.3)
openssl (2.0.2 < 2.0.3)
power_assert (0.4.1 < 1.0.1)
xmlrpc (0.2.1 < 0.3.0)
leandro@samsara:~ $ gem update
Updating installed gems
Updating bigdecimal
Fetching: bigdecimal-1.3.1.gem (100%)
Building native extensions.  This could take a while...
Successfully installed bigdecimal-1.3.1
Updating json
Fetching: json-2.0.3.gem (100%)
Building native extensions.  This could take a while...
Successfully installed json-2.0.3
Updating openssl
Fetching: openssl-2.0.3.gem (100%)
Building native extensions.  This could take a while...
Successfully installed openssl-2.0.3
Updating power_assert
Fetching: power_assert-1.0.1.gem (100%)
Successfully installed power_assert-1.0.1
Updating xmlrpc
Fetching: xmlrpc-0.3.0.gem (100%)
Successfully installed xmlrpc-0.3.0
Gems updated: bigdecimal json openssl power_assert xmlrpc
leandro@samsara:~ $ gem cleanup
Cleaning up installed gems...
Clean Up Complete
leandro@samsara:~ $ gem install bundler
Fetching: bundler-1.14.4.gem (100%)
Did you know that maintaining and improving Bundler and RubyGems.org costs more than $25,000 USD every month? Help us keep the gem ecosystem free for everyone by joining the hundreds of companies and individuals who help cover these costs: https://ruby.to/support-bundler
Successfully installed bundler-1.14.4
1 gem installed
leandro@samsara:~ $ gem install rails
Fetching: i18n-0.8.0.gem (100%)
Successfully installed i18n-0.8.0
Fetching: thread_safe-0.3.5.gem (100%)
Successfully installed thread_safe-0.3.5
Fetching: tzinfo-1.2.2.gem (100%)
Successfully installed tzinfo-1.2.2
Fetching: concurrent-ruby-1.0.4.gem (100%)
Successfully installed concurrent-ruby-1.0.4
Fetching: activesupport-5.0.1.gem (100%)
Successfully installed activesupport-5.0.1
Fetching: rack-2.0.1.gem (100%)
Successfully installed rack-2.0.1
Fetching: rack-test-0.6.3.gem (100%)
Successfully installed rack-test-0.6.3
Fetching: mini_portile2-2.1.0.gem (100%)
Successfully installed mini_portile2-2.1.0
Fetching: nokogiri-1.7.0.1.gem (100%)
Building native extensions.  This could take a while...
Successfully installed nokogiri-1.7.0.1
Fetching: loofah-2.0.3.gem (100%)
Successfully installed loofah-2.0.3
Fetching: rails-html-sanitizer-1.0.3.gem (100%)
Successfully installed rails-html-sanitizer-1.0.3
Fetching: rails-dom-testing-2.0.2.gem (100%)
Successfully installed rails-dom-testing-2.0.2
Fetching: builder-3.2.3.gem (100%)
Successfully installed builder-3.2.3
Fetching: erubis-2.7.0.gem (100%)
Successfully installed erubis-2.7.0
Fetching: actionview-5.0.1.gem (100%)
Successfully installed actionview-5.0.1
Fetching: actionpack-5.0.1.gem (100%)
Successfully installed actionpack-5.0.1
Fetching: activemodel-5.0.1.gem (100%)
Successfully installed activemodel-5.0.1
Fetching: arel-7.1.4.gem (100%)
Successfully installed arel-7.1.4
Fetching: activerecord-5.0.1.gem (100%)
Successfully installed activerecord-5.0.1
Fetching: globalid-0.3.7.gem (100%)
Successfully installed globalid-0.3.7
Fetching: activejob-5.0.1.gem (100%)
Successfully installed activejob-5.0.1
Fetching: mime-types-data-3.2016.0521.gem (100%)
Successfully installed mime-types-data-3.2016.0521
Fetching: mime-types-3.1.gem (100%)
Successfully installed mime-types-3.1
Fetching: mail-2.6.4.gem (100%)
Successfully installed mail-2.6.4
Fetching: actionmailer-5.0.1.gem (100%)
Successfully installed actionmailer-5.0.1
Fetching: nio4r-1.2.1.gem (100%)
Building native extensions.  This could take a while...
Successfully installed nio4r-1.2.1
Fetching: websocket-extensions-0.1.2.gem (100%)
Successfully installed websocket-extensions-0.1.2
Fetching: websocket-driver-0.6.5.gem (100%)
Building native extensions.  This could take a while...
Successfully installed websocket-driver-0.6.5
Fetching: actioncable-5.0.1.gem (100%)
Successfully installed actioncable-5.0.1
Fetching: thor-0.19.4.gem (100%)
Successfully installed thor-0.19.4
Fetching: method_source-0.8.2.gem (100%)
Successfully installed method_source-0.8.2
Fetching: railties-5.0.1.gem (100%)
Successfully installed railties-5.0.1
Fetching: sprockets-3.7.1.gem (100%)
Successfully installed sprockets-3.7.1
Fetching: sprockets-rails-3.2.0.gem (100%)
Successfully installed sprockets-rails-3.2.0
Fetching: rails-5.0.1.gem (100%)
Successfully installed rails-5.0.1
35 gems installed
leandro@samsara:~ $ ruby -v
ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin15]
leandro@samsara:~ $ gem -v
2.6.10
leandro@samsara:~ $ rails -v
Rails 5.0.1
leandro@samsara:~ $ gem list

*** LOCAL GEMS ***

actioncable (5.0.1)
actionmailer (5.0.1)
actionpack (5.0.1)
actionview (5.0.1)
activejob (5.0.1)
activemodel (5.0.1)
activerecord (5.0.1)
activesupport (5.0.1)
arel (7.1.4)
bigdecimal (1.3.1, default: 1.3.0)
builder (3.2.3)
bundler (1.14.4)
bundler-unload (1.0.2)
concurrent-ruby (1.0.4)
did_you_mean (1.1.0)
erubis (2.7.0)
executable-hooks (1.3.2)
gem-wrappers (1.2.7)
globalid (0.3.7)
i18n (0.8.0)
io-console (default: 0.4.6)
json (2.0.3, default: 2.0.2)
loofah (2.0.3)
mail (2.6.4)
method_source (0.8.2)
mime-types (3.1)
mime-types-data (3.2016.0521)
mini_portile2 (2.1.0)
minitest (5.10.1)
net-telnet (0.1.1)
nio4r (1.2.1)
nokogiri (1.7.0.1)
openssl (2.0.3, default: 2.0.2)
power_assert (1.0.1, 0.4.1)
psych (default: 2.2.2)
rack (2.0.1)
rack-test (0.6.3)
rails (5.0.1)
rails-dom-testing (2.0.2)
rails-html-sanitizer (1.0.3)
railties (5.0.1)
rake (12.0.0)
rdoc (default: 5.0.0)
rubygems-bundler (1.4.4)
rvm (1.11.3.9)
sprockets (3.7.1)
sprockets-rails (3.2.0)
test-unit (3.2.3)
thor (0.19.4)
thread_safe (0.3.5)
tzinfo (1.2.2)
websocket-driver (0.6.5)
websocket-extensions (0.1.2)
xmlrpc (0.3.0, 0.2.1)
leandro@samsara:~ $ rvm version
rvm 1.29.0 (bugfix/3928_dynamic_openssl_check_for_ruby24) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io/]
```
**Thank you very much for your effort!**
